File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        
        summ = sum(chalk);
        k = k%summ

        for i in range(len(chalk)):
            k = k - chalk[i]
            if k < 0:
                return i

File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.java
class Solution {
    public int chalkReplacer(int[] chalk, int k) {
        
        long sum = 0;
        for(int i=0;i<chalk.length;i++){
            sum += (long)chalk[i];
        }
        k = (int)( (long)k%sum );
//         temporarily making k as long evaluating the exprn then making it back int
        
        for(int i = 0;i<chalk.length;i++){
            k=k-chalk[i];
            if (k<0)return i;
        }
        return -1;
    }
}